sebs/fission/config.py

```diff
@@ -0,0 +1,313 @@
+        try:
+            fission_env_list = subprocess.run(
+                "fission env list ".split(),
+                stdout=subprocess.PIPE,
+                stderr=subprocess.DEVNULL,
+            )
+            subprocess.run(
+                f"grep {name}".split(),
+                check=True,
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+                input=fission_env_list.stdout,
+            )
+            # PK: Add Logger 
+            print(f"Env {name} already exist")
+            # ret.logging.info(f"Env {name} already exist")
+        except subprocess.CalledProcessError:
+            # PK: Add Logger
+            # ret.logging.info(f'Creating env for {name} using image "{image}".')
+            print(f'Creating env for {name} using image "{image}".')
+            try:
+                connection_uri = runtime_env[2]
+                access_key = runtime_env[1]
+                secret_key = runtime_env[0]
+                subprocess.run(
+                    f"fission env create --name {name} --image {image} \
+                    --builder {builder} --runtime-env {runtime_env[2]} --runtime-env {runtime_env[1]} --runtime-env {runtime_env[0]}".split(),
+                    check=True,
+                    stdout=subprocess.DEVNULL,
+                ) 
+                print(f"Successfully created the enviroment {name}")
+            except subprocess.CalledProcessError:
+                print(f"Creating env {name} failed.")
+                # PK: Add Logger
+                # logging.info(f"Creating env {name} failed. Retrying...")
+            # PK: Add Logger
+
+
+    def update_cache(self, cache: Cache):
+        super().update_cache(cache)
+        cache.update_config(
+            val=self.docker_registry, keys=["fission", "resources", "docker", "registry"]
+        )
+        cache.update_config(
+            val=self.docker_username, keys=["fission", "resources", "docker", "username"]
+        )
+        cache.update_config(
+            val=self.docker_password, keys=["fission", "resources", "docker", "password"]
+        )
+        if self._storage:
+            self._storage.update_cache(["fission", "resources", "storage"], cache)
+
+    def serialize(self) -> dict:
+        out: dict = {
+            **super().serialize(),
+            "docker_registry": self.docker_registry,
+            "docker_username": self.docker_username,
+            "docker_password": self.docker_password,
+        }
+        if self._storage:
+            out = {**out, "storage": self._storage.serialize()}
+        return out
+
+
+
+
+class FissionConfig(Config):
+    name: str
+    cache: Cache
+    shouldShutdown: bool
+
+    def __init__(self, config: dict, cache: Cache):
+        super().__init__(name="fission")
+        self._credentials = FissionCredentials()
+        self._resources = FissionResources()
+        self.shutdownStorage = config["shutdownStorage"]
```

Comment:
You can remove shutdown storage and remove cluster - those options are no longer needed and we don't have to support them